implementations/rust/ockam/ockam_node/src/storage/database/sqlx_database.rs

```diff
@@ -308,67 +311,110 @@ impl SqlxDatabase {
         })
     }
 
-    pub(crate) async fn create_connection_pool(
-        configuration: &DatabaseConfiguration,
-    ) -> Result<Pool<Any>> {
-        install_default_drivers();
-        let connection_string = configuration.connection_string();
-        debug!("connecting to {connection_string}");
+    fn pool_options(configuration: &DatabaseConfiguration) -> Result<PoolOptions<Any>> {
+        let database_type = configuration.database_type();
 
-        let options = AnyConnectOptions::from_str(&connection_string)
-            .map_err(Self::map_sql_err)?
-            .log_statements(configuration.statements_log_level())
-            .log_slow_statements(LevelFilter::Trace, Duration::from_secs(1));
-
-        // sqlx default is 10, 16 is closer to the typical number of threads spawn
-        // by tokio within a node, but has no particular reason
-        const MAX_POOL_SIZE: u32 = 16;
-
-        let max_pool_size = match configuration.mode() {
-            DatabaseConfigurationMode::SqlitePersistent {
-                single_connection, ..
-            }
-            | DatabaseConfigurationMode::SqliteInMemory { single_connection } => {
-                if *single_connection {
-                    1
-                } else {
-                    MAX_POOL_SIZE
-                }
-            }
-            _ => MAX_POOL_SIZE,
+        let max_pool_size = if configuration.is_single_connection() {
```

Comment:
I think it would be better to get the pool size from the environment variable directly when creating the configuration and not get it from this part of the code.